homeassistant/components/unifiprotect/config_flow.py

```diff
@@ -111,6 +124,7 @@ async def async_step_integration_discovery(
                     not entry_has_direct_connect
                     and is_ip_address(entry_host)
                     and entry_host != source_ip
+                    and _async_entry_is_in_failure_state(self.hass, entry)
                 ):
                     new_host = source_ip
```

Comment:
```suggestion
                    if await _async_is_bad_protect_host(entry_host):
                        new_host = source_ip
```

Reply:
AFAICT its not possible to get a discovery in this state
https://github.com/bdraco/unifi-discovery/blob/4d369813200993222640394457c2ef72dbdf7f95/unifi_discovery/__init__.py#L432

Reply:
The suggestion is to check that the current ip is also unreachable by probing the system endpoint.

That needs to be done in the discovery code though so it makes this change much more complex and will have to wait for another day.